internal/utils/comparators/deployments/comparator_test.go

```diff
@@ -58,6 +59,15 @@ var _ = Describe("deployments#AreSame", func() {
 			ok, _ := deployments.AreSame(current, desired)
 			Expect(ok).To(BeTrue())
 		})
+
+		It("should fail when the secret hash is different", func() {
+			current.Spec.Template.Annotations = map[string]string{
+				constants.AnnotationSecretHash: "foo",
+			}
+			ok, _ := deployments.AreSame(current, desired)
+
+			Expect(ok).To(BeFalse())
```

Comment:
Similar to `deployments`:

```suggestion
			ok, reason := deployments.AreSame(current, desired)

			Expect(ok).To(BeFalse())
			Expect(reason).To(Equal("secretHash"))
```